platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -64,47 +68,157 @@
   private final int maximumShardsPerNode =
       NumberUtils.toInt(System.getProperty("solr.cloud.maxShardPerNode"), 2);
 
+  private static final String ALIAS_PROP = "alias";
+
   @Override
   public org.codice.solr.client.solrj.SolrClient newClient(String core) {
+    return newClient(core, null);
+  }
+
+  @Override
+  public org.codice.solr.client.solrj.SolrClient newClient(
+      String collection, Map<String, Object> properties) {
+
     String zookeeperHosts = System.getProperty("solr.cloud.zookeeper");
+    checkConfig(zookeeperHosts);
 
-    if (StringUtils.isBlank(zookeeperHosts)) {
-      LOGGER.warn(
-          "Cannot create SolrCloud client without Zookeeper host list system property [solr.cloud.zookeeper] being set.");
-      throw new IllegalStateException("system property 'solr.cloud.zookeeper' is not configured");
-    }
     LOGGER.debug(
-        "Solr({}): Creating a SolrCloud client using Zookeeper hosts [{}]", core, zookeeperHosts);
+        "Solr({}): Creating a Solr Cloud client with configuration using Zookeeper hosts [{}]",
+        collection,
+        zookeeperHosts);
+
+    String alias = properties == null ? null : (String) properties.getOrDefault(ALIAS_PROP, null);
```

Comment:
✏️ Just an opinion, firming up the null contracts with the properties parameter could clean up null checks like this and reduce it down to just the getOrDefault call.